src/Tools/Test Framework/Test Runner/src/TestSuiteMgt.Codeunit.al

```diff
@@ -421,48 +435,68 @@ codeunit 130456 "Test Suite Mgt."
+            Selection := StrMenu(this.SelectCodeunitsToRunQst, 1)
         else
-            Selection := StrMenu(SelectTestsToRunQst, 3);
+            Selection := StrMenu(this.SelectTestsToRunQst, 3);
 
         exit(Selection);
     end;
 
-    local procedure GetLineNoFilter(TestMethodLine: Record "Test Method Line"; Selection: Option) LineNoFilter: Text
+    local procedure GetLineNoFilter(TestMethodLine: Record "Test Method Line"; Selection: Option): Text
     var
         DummyALTestSuite: Record "AL Test Suite";
-        CodeunitTestMethodLine: Record "Test Method Line";
-        MinNumber: Integer;
     begin
-        LineNoFilter := '';
         case Selection of
             DummyALTestSuite."Run Type"::"Active Test":
-                begin
-                    TestMethodLine.TestField("Line Type", TestMethodLine."Line Type"::"Function");
-                    LineNoFilter := Format(TestMethodLine."Line No.");
-                    CodeunitTestMethodLine.SetRange("Test Suite", TestMethodLine."Test Suite");
-                    CodeunitTestMethodLine.SetRange("Test Codeunit", TestMethodLine."Test Codeunit");
-                    CodeunitTestMethodLine.FindFirst();
-#pragma warning disable AA0217                    
-                    LineNoFilter := StrSubstNo('%1|%2', CodeunitTestMethodLine."Line No.", TestMethodLine."Line No.");
-#pragma warning restore
-                end;
+                exit(this.GetLineNoFilterActiveTest(TestMethodLine));
             DummyALTestSuite."Run Type"::"Active Codeunit":
-                begin
-                    CodeunitTestMethodLine.SetRange("Test Suite", TestMethodLine."Test Suite");
-                    CodeunitTestMethodLine.SetRange("Test Codeunit", TestMethodLine."Test Codeunit");
-                    CodeunitTestMethodLine.SetAscending("Line No.", true);
-                    CodeunitTestMethodLine.FindFirst();
-                    MinNumber := CodeunitTestMethodLine."Line No.";
-                    CodeunitTestMethodLine.FindLast();
-#pragma warning disable AA0217
-                    LineNoFilter :=
-                      StrSubstNo('%1..%2', MinNumber, CodeunitTestMethodLine."Line No.");
-#pragma warning restore                      
-                end;
+                exit(this.GetLineNoFilterForTestCodeunit(TestMethodLine));
         end;
     end;
 
+    local procedure GetLineNoFilterActiveTest(TestMethodLine: Record "Test Method Line"): Text
+    var
+        CodeunitTestMethodLine: Record "Test Method Line";
+        LineNoFilter: Text;
+    begin
+        TestMethodLine.TestField("Line Type", TestMethodLine."Line Type"::"Function");
+        CodeunitTestMethodLine.SetRange("Test Suite", TestMethodLine."Test Suite");
+        CodeunitTestMethodLine.SetRange("Test Codeunit", TestMethodLine."Test Codeunit");
+        CodeunitTestMethodLine.SetFilter("Line No.", this.GetLineNoFilterForTestCodeunit(TestMethodLine));
+        CodeunitTestMethodLine.FindFirst();
```

Comment:
Might be over the top, but suggesting it anyway:
```suggestion
        CodeunitTestMethodLine.SetFilter("Line No.", this.GetLineNoFilterForTestCodeunit(TestMethodLine));
        CodeunitTestMethodLine.SetLoadfields("Line No.");
        CodeunitTestMethodLine.FindFirst();
```

Reply:
SetLoadFields make no sense in this tool. 
SetLoadFields only speed up the reading of the records when there are many table extensions installed, it was the issue in the past.

SQL read of this table will be blazingly fast. JIT load will be slow if we do SetLoadDields.